docs: add visualization approach decision guidance & improve intros to balance both modes (#5447) (#6560)
elastic/docs-content: explore-analyze/visualize/esorql.md

# ES|QL visualizations [esql-visualizations]

> {{esql}} mode is particularly useful when you need to:

Author: ```suggestion
Creating visualizations using queries with {{esql}} mode is particularly useful when you need to:
```

> Creating visualizations using an {{esql}} query is particularly useful when you need to:

Reviewer: This is nice.

- Query data across multiple indices without a pre-configured data view
- Apply complex filtering, transforms, or custom calculations in a single query
- Prototype a visualization directly from an {{esql}} query in Discover

> For simpler aggregations on a known index, [Lens in point-and-click mode](lens.md) may be easier to use.

Author: ```suggestion
For less advanced aggregations on a known index, [the point-and-click mode](lens.md) is a good alternative.
```
